### PR TITLE
feat(unlock-js): Migrate to typescript

### PR DESCRIPTION
**Pull request by searchableguy**
# Description

Work in progress typescript migration

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Comment by searchableguy** [comment]
Should we convert files in PublicLock, Unlock, and __tests___, etc folder to typescript as well?

To me, it doesn't seem it's worth it. We should write new files in typescript but leave it as it is because it won't benefit an external user. All external users will use our UnlockService which provides abstraction over those functions with types.

@clemsos  @julien51 WDYT?

File: packages/unlock-js/src/walletService.ts [hunk 1]
```diff
@@ -63,7 +65,7 @@ export default class WalletService extends UnlockService {
    */
   // eslint-disable-next-line no-underscore-dangle
   // TODO: Do we need this???
-  async _handleMethodCall(methodCall) {
+  async _handleMethodCall(methodCall: any) {
```
Comment (kalidiagne) [inline packages/unlock-js/src/walletService.ts:68]: this is a Promise, in the line below we are using await on `methodCall`
```suggestion
  async _handleMethodCall(methodCall: Promise<any>) {
```

Comment (searchableguy) [reply packages/unlock-js/src/walletService.ts:68]: Can also be a non promise. The distinction doesn't matter when you use any.

File: packages/unlock-js/src/web3Service.ts [hunk 2]
```diff
@@ -99,17 +105,17 @@ export default class Web3Service extends UnlockService {
    * and formats it to a string of ether.
    * Returns a promise with the balance
    */
-  async getAddressBalance(address, network) {
+  async getAddressBalance(address: string, network: number) {
     const balance = await this.providerForNetwork(network).getBalance(address)
-    return utils.fromWei(balance, 'ether')
+    return utils.fromWei(balance, 'ethers')
```
[the remark this answers is not in the document]
Comment (searchableguy) [reply packages/unlock-js/src/web3Service.ts:110]: Thanks. Make sense.

**Review by kalidiagne — APPROVED** [review]

**Comment by julien51** [comment]
Agreed. If it's not external, probably not worth doing right now.

File: locksmith/src/routes/index.ts [hunk 3]
```diff
@@ -22,8 +22,7 @@ router.use((request, _, next) => {
     // When the route starts with the chain (deprecated?)
     chain = parseInt(match[1])
   } else if (request.query?.chain) {
-    // @ts-expect-error
-    chain = parseInt(request.query.chain)
+    chain = parseInt(String(request.query.chain))
```
Comment (kalidiagne) [inline locksmith/src/routes/index.ts:25]: @searchableguy there is a possibility that this gives `NaN` if `request.query.chain` has a value thas is not a number 
```suggestion
    const value = String(request.query.chain)
    chain = !isNaN(value) ? parseInt(value) : undefined
```

what do you think?

Comment (searchableguy) [reply locksmith/src/routes/index.ts:25]: Looks okay.

**Review by julien51 — APPROVED** [review]

**Pull request merged**